hummingbot/strategy/pure_market_making/staggered_multiple_size_sizing_delegate.pyx

```diff
@@ -65,22 +70,40 @@ cdef class StaggeredMultipleSizeSizingDelegate(OrderSizingDelegate):
 
         for idx in range(self.number_of_orders):
             current_order_size = self.order_start_size + self.order_step_size * idx
-            required_quote_asset_balance += ( float(current_order_size) * float(pricing_proposal.buy_order_prices[idx]) )
-            required_base_asset_balance += float(current_order_size)
+            buy_fees = market.c_get_fee(market_info.base_currency, market_info.quote_currency,
+                                    OrderType.MARKET, TradeType.BUY,
+                                    current_order_size, pricing_proposal.buy_order_prices[idx])
+
             if market.name == "binance":
-                current_order_size = market.c_quantize_order_amount(market_info.symbol, current_order_size, pricing_proposal.buy_order_prices[idx])
+                #For binance fees is calculated in base token, so need to adjust for that
+                buy_order_size = market.c_quantize_order_amount(market_info.symbol, current_order_size, pricing_proposal.buy_order_prices[idx])
+                #Check whether you have enough quote tokens
+                required_quote_asset_balance += ( float(buy_order_size) * float(pricing_proposal.buy_order_prices[idx]) )
+
             else:
-                current_order_size = market.c_quantize_order_amount(market_info.symbol, current_order_size)
+                buy_order_size = market.c_quantize_order_amount(market_info.symbol, current_order_size)
+                #For other exchanges, fees is calculated in quote tokens, so need to ensure you have enough for order + fees
+                required_quote_asset_balance += ( float(buy_order_size) * float(pricing_proposal.buy_order_prices[idx]) *(1+float(buy_fees.percent)) )
```

Comment:
same comment as above

Reply:
Just to make sure that this is clear, the issue is NOT that we need to adjust the order size to reflect fees, either on the base or the quote side. 

The issue is that because Binance takes fees out of the base asset amount when you execute a buy order, you end up with a mismatched amount than expected if you are doing cross-exchange or arbitrage trades. Originally, we incremented the buy order size for Binance orders to make the amounts line up. However, this creates bugs and is confusing, so we are not doing this anymore. Other exchanges do not need any adjustment.

Reply:
Same comment as above.